File: problems/014/014.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* Copyright 2021 Melwyn Francis Carlo */

int main()
{
    int n_count = 1E5;

    int max_count = 0;
    int max_count_number = 0;

    while (n_count < 1E6)
    {
        int i_count = 0;
        long int n = n_count;

        while (n != 1)
        {
            if ((n % 2) == 0)
            {
                n /= 2;
            }
            else
            {
                n = (3 * n) + 1;
            }

            i_count++;
        }

        i_count++;

        if (i_count > max_count)
        {
            max_count        = i_count;
            max_count_number = n_count;
        }

        n_count++;
    }

    printf("%d\n", max_count_number);

    return 0;
}

File: problems/014/014.f90
PROGRAM F014

    ! Copyright 2021 Melwyn Francis Carlo

    IMPLICIT NONE

    INTEGER :: N_COUNT = 1E5

    INTEGER :: MAX_COUNT        = 0
    INTEGER :: MAX_COUNT_NUMBER = 0

    INTEGER :: I_COUNT

    INTEGER (KIND=8) :: N

    DO WHILE (N_COUNT < 1E6)

        I_COUNT = 0
        N = N_COUNT;

        DO WHILE (N /= 1)

            IF (MOD(N, INT(2, 8)) == 0) THEN
                N = N / 2;
            ELSE
                N = (3 * N) + 1;
            END IF

            I_COUNT = I_COUNT + 1

        END DO

        I_COUNT = I_COUNT + 1

        IF (I_COUNT > MAX_COUNT) THEN
            MAX_COUNT        = I_COUNT;
            MAX_COUNT_NUMBER = N_COUNT;
        END IF

        N_COUNT = N_COUNT + 1

    END DO

    PRINT ('(I0)'), MAX_COUNT_NUMBER

END PROGRAM F014
